### PR TITLE
Practice mode fixes & improvements

### PR DESCRIPTION
**Pull request by def-**
Thanks to NANAMAR for report.

Update: tested, works.

### CODE REVIEW

File: src/game/server/teams.cpp [hunk 1]
```diff
@@ -685,6 +685,7 @@ void CGameTeams::OnCharacterDeath(int ClientID, int Weapon)
 	else
 	{
 		SetForceCharacterTeam(ClientID, Team);
+		m_Practice[Team] = false;
```
Comment (heinrich5991) [inline src/game/server/teams.cpp:688]: This seems to be at the wrong place, should be further down where the team is guaranteed to be killed.

Comment (def-) [reply src/game/server/teams.cpp]: Done.

File: src/game/server/teams.cpp [hunk 2]
```diff
@@ -706,6 +706,12 @@ void CGameTeams::OnCharacterDeath(int ClientID, int Weapon)
 						GameServer()->SendChatTarget(i, aBuf);
 				}
 		}
+
+		m_Practice[Team] = false;
+
+		for(int i = 0; i < MAX_CLIENTS; i++)
+			if(m_Core.Team(i) == Team && GameServer()->m_apPlayers[i])
+				GameServer()->m_apPlayers[i]->m_VotedForPractice = false;
```
Comment (heinrich5991) [inline src/game/server/teams.cpp:714]: What I meant was that this should go into the `if` clause. Code here (and above the `if` clause) also gets executed when not all of the team's members are killed.

Comment (def-) [reply src/game/server/teams.cpp:714]: I tried that initially but then it doesn't reset for a locked single player team.

Comment (Learath2) [reply src/game/server/teams.cpp:714]: But as is wouldn't this cause teams to get out of practice mode if any tee dies?

Comment (def-) [reply src/game/server/teams.cpp:717]: Only if the team is locked. And locked teams get every player killed. Or do you mean before the start line?

Comment (def-) [reply src/game/server/teams.cpp:717]: I guess it makes sense to only reset after the start line, ok, will change.

**Comment by def-** [comment]
Good to go I guess

**Comment by def-** [comment]
Can we get this in? I feel like the current state of practice is not so great and this would improve it.

**Comment by Learath2** [comment]
LGTM

bors r+

**Pull request merged**